sycl/doc/extensions/proposed/sycl_ext_intel_fpga_pipes_properties.asciidoc

```diff
@@ -134,6 +134,11 @@ struct uses_ready {
   using value_t = property_value<uses_ready, std::integral_constant<bool, Ready>>;
 };
 
+struct in_csr {
+  template<bool Csr>
+  using value_t = property_value<in_csr, std::integral_constant<bool, Csr>>;
+};
+
```

Comment:
Can we potentially use the same in_csr property defined in [device global](https://github.com/intel/llvm/blob/7c8a2ace91d370c12e0b2e9e96c4544c7ef3108d/sycl/doc/extensions/proposed/sycl_ext_oneapi_device_global.asciidoc#L561)? Given the behaviour will be similar

Reply:
I think we could define the property in both extensions, so you could use it with either one.  You would not need to enable the device global extension just to use the property.

Note that there is a proposal to rename the "implement_in_csr" property like this:

```
enum class hw_interface_enum : /* unspecified */ {
  csr,
  dedicated
};

struct hw_interface_key {
  template<hw_interface_enum Interface>
  using value_t =
      property_value<hw_interface_key, std::integral_constant<int, Interface>>;
};

template <hw_interface_enum Interface>
inline constexpr hw_interface_key::value_t<Interface> hw_interface;
inline constexpr hw_interface_key::value_t<hw_interface_enum::csr>
    hw_interface_csr;
inline constexpr hw_interface_key::value_t<hw_interface_enum::dedicated>
    hw_interface_dedicated;
```

Does that new spelling still make sense for pipes?

Reply:
I think it makes sense, and it also makes this property more extensible.

Reply:
I think "dedicated" isn't quite appropriate for host pipes. We can and will share a single hardware pipe with multiple logical pipes.

Reply:
Can you propose a different name that makes sense for both the pipe case and the device global case?

Reply:
Nothing comes to mind immediately (but that doesn't mean there isn't one that makes sense for both). @GarveyJoe do you have any ideas?

Reply:
Adding myself

Reply:
"conduit" or "streaming" could both work.  Alternatively, we could go back to it being a bool.

Reply:
I could see either of those for pipe, but it's less clear to me they're descriptive of device_global (@artemrad ?) For bool, how confident are we won't eventually want another option?

Reply:
Should in_csr be a separate protocol?  It isn't an avalon-st interface anymore.  It's closer to an avalon-mm interface.  Some of the protocol-dependent properties don't even make sense for an "in_csr" pipe such as ready_latency.